sdk/js/src/Client.ts

```diff
@@ -191,7 +198,7 @@ export class DVCClient implements Client {
             this.eventQueue.flushEvents()
 
             let updatedUser: DVCPopulatedUser
-            const bothAnonymous = this.user.isAnonymous && user.isAnonymous
+            const bothAnonymous = this.user.isAnonymous && (user.isAnonymous || true)
```

Comment:
(user.isAnonymous || true) always evaluates to true

Reply:
ah, this should be the same as the User constructor `isAnonymous`.. will update to `(user.isAnonymous || !user.user_id)`